### PR TITLE
Global store for sync manager / smart sync plugin

### PR DESCRIPTION
**Pull request by wmathurin**
Sync manager now only needs a smartstore to be instantiated (but still needs to have a rest client to run syncs)
New js tests hooked up and passing

### CODE REVIEW

File: libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java [hunk 1]
```diff
@@ -108,62 +131,40 @@ public static synchronized SyncManager getInstance(UserAccount account, String c
...
-            instance = new SyncManager(account, communityId);
-            INSTANCES.put(uniqueId, instance);
-        }
-        return instance;
+
+        SmartStore smartStore = SmartSyncSDKManager.getInstance().getSmartStore(account, communityId);
+        SyncManager syncManager = getInstanceForStore(smartStore);
+
+        // Setting up rest client
+        RestClient restClient = SalesforceSDKManager.getInstance().getClientManager().peekRestClient(account);
```
Comment (khawkins) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java]: If I'm reading this right, if you get an instance for a user, you set the RestClient, but if you go straight to getting an instance for a store, you don't?  How does that play out across the different usage scenarios of `SyncManager`?

Comment (khawkins) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java]: Presumably, the developer would manually specify the RestClient for that manager, in that case?

**Comment by khawkins** [comment]
Looks fine to me, over all.

File: libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java (hunk 1 above)
Comment (wmathurin) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java]: Alternatively, we could change the signature for the sync methods and add a rest client parameter there.
On the iOS side, SFRestAPI is a singleton and can be obtained from anywhere, and therefore the sync manager doesn't have to concern itself with it.

BTW another thing we need to figure out on the Android side is unauthenticated calls.

Comment (khawkins) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java]: Makes sense. :+1:

**Comment by wmathurin** [comment]
Will revisit SyncManager instantiation in a future pull request

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**